scripts/parallel.rb

```diff
@@ -4,58 +4,105 @@
-  rescue
-      driver.execute_script('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed", "reason": "Some elements failed to load"}}')
-  end
-  driver.quit 
+def run_session(capabililties)
+    options = Selenium::WebDriver::Options.send capabililties['browserName']
+    options.browser_version = capabililties['browserVersion'] if capabililties['browserVersion']
+    options.add_option('bstack:options', capabililties['bstack:options'])
+
+    driver = Selenium::WebDriver.for(
+        :remote,
+        :url => "https://#{USER_NAME}:#{ACCESS_KEY}@hub.browserstack.com/wd/hub",
+        :capabilities => options
+    )
+    begin
+        # opening the bstackdemo.com website
+        driver.navigate.to "https://bstackdemo.com"
+        wait = Selenium::WebDriver::Wait.new(:timeout => 10) # seconds
+        wait.until { !driver.title.match(/StackDemo/i).nil? }
+
+        # getting name of the product available on the webpage
+        product = driver.find_element(:xpath, '//*[@id="1"]/p')
+        wait.until { product.displayed? }
+        product_text = product.text
+
+        # wait until the Add to Cart button is displayed on webpage, then click it
+        cart_btn = driver.find_element(:xpath, '//*[@id="1"]/div[4]')
+        wait.until { cart_btn.displayed? }
+        cart_btn.click
+
+        # waiting until the Cart pane appears
+        wait.until { 
+            driver.find_element(
+                :xpath, 
+                '//*[@id="__next"]/div/div/div[2]/div[2]/div[2]/div/div[3]/p[1]'
+            ).displayed? 
+        }
+
+        # get name of the product in the cart
+        product_in_cart = driver.find_element(
+            :xpath, 
+            '//*[@id="__next"]/div/div/div[2]/div[2]/div[2]/div/div[3]/p[1]'
+        )
+        wait.until { product_in_cart.displayed? }
+        product_in_cart_text = product_in_cart.text
+
+        # check if the product has been added to the cart
+        if product_text.eql? product_in_cart_text
+            # mark test as 'passed' if the product is successfully added to the cart
+            driver.execute_script(
+                'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed", "reason": "Product has been successfully added to the cart!"}}'
+            )
+        else
+            # mark test as 'failed' if the product is not added to the cart
+            driver.execute_script(
+                'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed", "reason": "Failed to add product to the cart"}}'
+            )
+        end
+    rescue
+        # mark test as 'failed' if test script is unable to open the bstackdemo.com website
+        driver.execute_script(
+            'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed", "reason": "Some elements failed to load"}}'
```

Comment:
Can we pass exception message while marking session status in all the three tests (local, parallel, single) ?